test/mini_s3_tests.erl

```diff
@@ -19,112 +19,170 @@
 -module(mini_s3_tests).
 
 -include_lib("eunit/include/eunit.hrl").
+-include_lib("erlcloud/include/erlcloud_aws.hrl").
 -include("../src/internal.hrl").
 
+-spec format_s3_uri(aws_config(), string()) -> string().
+format_s3_uri(Config, Bucket) ->
+    erlcloud_s3:get_object_url(Bucket, "", Config).
+
 format_s3_uri_test_() ->
     Config = fun(Url, Type) ->
-                     #config{s3_url = Url, bucket_access_type = Type}
+                     mini_s3:new("", "", Url, Type)
              end,
     Tests = [
              %% hostname
-             {"https://my-aws.me.com", virtual_hosted, "https://bucket.my-aws.me.com:443"},
-             {"https://my-aws.me.com", path, "https://my-aws.me.com:443/bucket"},
+             {"https://my-aws.me.com", vhost, "https://bucket.my-aws.me.com:443/"},
+             {"https://my-aws.me.com", path,  "https://my-aws.me.com:443/bucket/"},
 
              %% ipv4
-             {"https://192.168.12.13", path, "https://192.168.12.13:443/bucket"},
+             {"https://192.168.12.13", path,  "https://192.168.12.13:443/bucket/"},
 
              %% ipv6
              {"https://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]", path,
-              "https://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:443/bucket"},
+              "https://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:443/bucket/"},
 
              %% These tests document current behavior. Using
-             %% virtual_hosted with an IP address does not make sense,
+             %% vhost with an IP address does not make sense,
              %% but leaving as-is for now to avoid adding the
              %% is_it_an_ip_or_a_name code.
-             {"https://192.168.12.13", virtual_hosted, "https://bucket.192.168.12.13:443"},
+             {"https://192.168.12.13", vhost, "https://bucket.192.168.12.13:443/"},
 
-             {"https://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]", virtual_hosted,
-              "https://bucket.[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:443"}
+             {"https://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]", vhost,
+              "https://bucket.[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:443/"}
             ],
-    [ ?_assertEqual(Expect, mini_s3:format_s3_uri(Config(Url, Type), "bucket"))
+    [ ?_assertEqual(Expect, format_s3_uri(Config(Url, Type), "bucket"))
       || {Url, Type, Expect} <- Tests ].
 
-%% 2015-01-27 00:00:00 = 1422316800 = ?MIDNIGHT
--define(MIDNIGHT, 1422316800).
--define(DAY, 86400).
--define(HOUR, 3600).
-
-expiration_time_test_() ->
```

Comment:
I'm worried that we are removing these tests as I don't see what other tests replace them in terms of testing the main invariants of our cachable URLs